integrations/langchain-memgraph/langchain_memgraph/tools.py

```diff
@@ -0,0 +1,285 @@
+"""Memgraph tools."""
+
+from typing import Any, Dict, List, Optional, Type
+
+from langchain_core.callbacks import (
+    CallbackManagerForToolRun,
+)
+from langchain_core.tools import BaseTool
+from pydantic import BaseModel, ConfigDict, Field
+
+from core.api.memgraph import MemgraphClient
+from core.tools.schema import ShowSchemaInfoTool
+from core.tools.cypher import CypherTool
+from core.tools.page_rank import PageRankTool
+from core.tools.storage import ShowStorageInfoTool
+from core.tools.constraint import ShowConstraintInfoTool
+from core.tools.index import ShowIndexInfoTool
+from core.tools.config import ShowConfigTool
+from core.tools.trigger import ShowTriggersTool
+from core.tools.betweenness_centrality import BetweennessCentralityTool
+
+
+class BaseMemgraphTool(BaseModel):
+    """
+    Base tool for interacting with Memgraph.
+    """
+
+    db: MemgraphClient = Field(exclude=True)
+
+    model_config = ConfigDict(
+        arbitrary_types_allowed=True,
+    )
+
+
+class _QueryMemgraphToolInput(BaseModel):
+    """
+    Input query for Memgraph Query tool.
+    """
+
+    query: str = Field(..., description="The query to be executed in Memgraph.")
+
+
+class RunQueryMemgraphTool(BaseMemgraphTool, BaseTool):
+    """Tool for querying Memgraph."""
+
+    name: str = CypherTool(db=None).get_name()
+    """The name that is passed to the model when performing tool calling."""
+
+    description: str = CypherTool(db=None).get_description()
+    """The description that is passed to the model when performing tool calling."""
+
+    args_schema: Type[BaseModel] = _QueryMemgraphToolInput
+    """The schema that is passed to the model when performing tool calling."""
+
+    def _run(
+        self,
+        query: str,
+        run_manager: Optional[CallbackManagerForToolRun] = None,
+    ) -> List[Dict[str, Any]]:
+        return CypherTool(
+            db=self.db,
+        ).call({"query": query})
+
+
+class RunShowSchemaInfoTool(BaseMemgraphTool, BaseTool):
```

Comment:
Here you have Run but not Memgraph in the name -> ShowSchemaInfoMemgraphTool...also, do we need Memgraph in those tool names if you're importing it from langchain_memgraph tools anyways (when using it)?

Reply:
I have reverted the Memgraph from example above.